addons/l10n_th/views/report_invoice.xml

```diff
@@ -0,0 +1,77 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <template id="report_invoice_document" inherit_id="account.report_invoice_document">
+        <xpath expr="//span[@t-field='o.partner_id.vat']" position="after">
+            <t t-if="o.company_id.account_fiscal_country_id.code == 'TH'" >
+                <span t-field="o.l10n_th_company_info" />
+                <span t-field="o.l10n_th_branch_number" />
+            </t>
+        </xpath>
+        <xpath expr="//h2" position="before">
+            <t t-set="use_custom_l10n_header" t-value="o.l10n_th_custom_header and o.company_id.account_fiscal_country_id.code == 'TH'"/>
```

Comment:
o.country_code might be shorter

Reply:
l10n_th_custom_header could be  a method instead.  I suppose there is a logic behind and it is not something the user needs to decide upon manually for each invoice.

Reply:
after consulting with the FC, she mentioned that the logical flow of this tax is a bit more complex. Some companies like to print the tax invoice first then the normal invoice, some prefer the reverse. In this way, it's way easier for them to choose which they like to print first through the use of this field